satpy/modifiers/angles.py

```diff
@@ -229,8 +229,12 @@ def _sanitize_observer_look_args(*args):
         if isinstance(arg, datetime):
             new_args.append(STATIC_EARTH_INERTIAL_DATETIME)
         elif isinstance(arg, (float, np.float64, np.float32)):
-            # round floating point numbers to nearest tenth
-            new_args.append(round(arg, 1))
+            # round large floating point numbers to nearest thousand
```

Comment:
Could be pass kwargs instead of args to this function to know what we are working on? Because this looks a bit like guessing what the argument is...

I think @djhoese wrote this function originally, so maybe he should be answering this question

Reply:
Ah there was a conversation on slack that was never recorded on here. My summary is that I don't think this aggressive rounding should be done here, but rather in the reader. This is definitely guessing what arguments are what, but that's why the rounding is relatively limited and is only meant to avoid floating point precision equality issues. That said, this is a sanitize function *specifically* for arguments going to the observer look function. I think I made this not kwargs because I didn't want to accidentally pass a keyword argument that could be interpreted by the `map_blocks` and also wanted to avoid kwargs in the function when they were required. I seem to recall there was some other issue with @Plantain's use where it didn't seem like the AHI HSD reader was rounding to the nearest 150m for altitude when it should have been.

So like I said, this rounding should be done in the AHI reader(s) and should be discussed on what the allowed precision is. That may lead to discussion of not doing the rounding on the `actual` position which @simonrp84 previously approved but that was with the misunderstanding that the original "actual" position was being retained. He may have even stronger feelings now that we've identified the misunderstanding *and* we want more aggressive rounding. I'm still not sure this aggressive of rounding is needed, but we'd need a real world data case to prove that the 150m rounding was happening and wasn't enough.

Reply:
I haven't yet tracked down exactly where or how the 150m rounding is meant to be happening, but I can confirm with satpy@git I am still rapidly accumulating sensor angle directories with different Himawari observation times.

Just adding print(arg) to the _sanitize_observer_look_args function shows for the following observation times:
202203250000: 140.636
202203251200: 140.65900000000002
202203260000: 140.64100000000002

Reply:
That was actually longitude, not altitude, so interesting to discover longitude has the same problem with drifting beyond it's rounding.
Altitude is actually:
202203250000: 35786850
202203251200: 35787450
202203260000: 35783700

Reply:
@djhoese Yes, I still think that if we're rounding stuff then the original should be retained. Am fine with rounding the `actual` position just as long as the user can still retrieve the original value if they desire :)

@plantain That's not unexpected. Himawari-8 is one of the least troublesome satellites for stationkeeping but it still moves around quite a bit. What's the rounding on lat/lon just now?

Reply:
As below, it could easily be rounded more aggressively but I have no idea if this is safe to do.

        if self._round_actual_position:
            actual_lon = round(actual_lon, 3)
            actual_lat = round(actual_lat, 2)
            actual_alt = round(actual_alt / 150) * 150  # to the nearest 150m

Reply:
@Plantain Can you confirm the git commit SHA you are using? The results you are getting suggest that `satellite_actual_altitude` is being rounded and used but the unrounded `nadir_longitude/latitude` are being used too. There was a PR I made that was merged a while ago that added a `preference` to `satpy/utils.py;get_satpos` which should be used in `satpy/modifiers/angles.py:_get_sensor_angles` and passed as "actual". If that isn't true in your version then you'll need to update your version of satpy.